test/kuzzle/constructor.test.js

```diff
@@ -750,6 +750,26 @@ describe('Kuzzle constructor', () => {
         });
       });
 
+      it('should get the url redirection from OAUTH', function (done) {
```

Comment:
Since we're going to add some code to handle arguments, I suggest adding another unit test checking for all argument combinations.